### DISCUSSION TITLE
FOSM and Dataworth Notebook suggestions

### DISCUSSION THREAD

**Discussion by laat0003** [gmdsi/GMDSI_notebooks#89] [General]
Hi,

I think the FOSM and Dataworth notebook could benefit from some comments around how the results may differ depending on the assumed spatial correlation length defined by the prescribed variogram. Specifically, where the plot of uncertainty reduction is made. I think that is a great spot to emphasize how the chosen variogram influences the flow of information from the observations to the pilot-points.  For example, how would the uncertainty reduction values at pilot-points differ if I change the variogram? At the extreme end, if I assume no spatial correlation, what happens to the uncertainty reductions then?

The other suggestion is a comment around the reductions to uncertainty with the notional calibration versus the actual reductions you get. I know you make a statement right at the end about the analysis depending on the assumption that you can fit the data according to the prescribed measurement noise. So, if the notional calibration informs me that calibration is worth pursuing, should I expect to get those same reductions moving forward? Could it be even better? How different could they be? What are some of the reasons why I may not achieve the same reductions that the notional demonstrated?

Now I know we already did the calibration in a previous notebook so we know what the results are but typically you wouldn't do it in that order. Also, some of the questions were addressed in the previous notebook, but it may be worthwhile reiterating them here.

Cheers,
Tariq

**Comment by jtwhite79** [comment]
Thanks Tariq.  Im kinda torn these days about dataworth and @mnfienen , @briochh,  @rhugman, @kmarkovich-intera, and I were recently discussing using dataworth in real-world settings.  Up until very recently, I had never actually seen dataworth in practice at a real site being used to either prune or expand a monitoring network...rather its usually just an exercise used in workshops (and notebooks!) to help build understanding about FOSM techniques.  However, I have used it a lot as a diagnostic when trying to understand unexpected (or erroneous) results, and I think it does have considerable power there.

For me, using FOSM to estimate the second moment of a posterior after calibration/history-matching is used to find the first moment is still a relevant use case for situations where you dont use ensembles.  But using FOSM in "pre-calibration" mode has struck me as sort "ok that fine, but Im stil going to calibrate the way I was planning to originally".  I wonder if people do use FOSM/dataworth to design an objective function that is more targeted to a specific prediction?  

I agree we could expand on how "hyper parameters" like the variogram propagate thru these analyses...pull request?!

Keen to hear experiences on this...

**Reply by laat0003** [reply]
Ha! You nailed it. The only time I've ever done data worth was to demonstrate it. Truth be told I haven't used FOSM much since 2019 unless explicitly asked to do so. I've generally defaulted to IES and stack-based optimization with OPT as my go-to. I find it is significantly easier to convey the UA side of things when it comes to reporting. Ensembles just seem to make sense to most people - even non-modelers. However, I do really like FOSM and want to use it more but unless I'm asked to redesign a model from scratch this is often impossible because in general, the existing models run so slow, that it feels like they are actually moving backwards through time. Hit run and I slowly transform to Vanilla Ice or MC-Hammer (showing my age here). Sure, you can just throw more cores at it but depending on the client that's not always an option. Perhaps my salvation rests with DSI? 

I haven't done any real testing on the effect that different variograms may have because I primarily used irregular pilot-point distributions in combination with John's MKPPSTAT and PPCOV_SVA prior to Pyemu becoming my default parameterization approach. I now just use the regular spacing via pp_space argument in add_parameters with DISV (thanks @rhugman ) or DISU. 

As for pull-requests - I think doing a survey of who uses Git (local) or with Github during the course might be useful. I've only just started learning how to use them despite adopting scripted modelling workflows for several years (hence no pull request from me...yet). I am sure that stepping through a tried and tested modelling workflow with Git and Github is something that will be well received by many. Although it may also produce a deluge of pull requests following the course.

**Reply by jtwhite79** [reply]
yeah good call on git(hub) usage...
